pkg/expr/transform.go

```diff
@@ -44,28 +41,15 @@ type Request struct {
 type Query struct {
 	RefID         string
 	TimeRange     TimeRange
-	DatasourceUID string        // deprecated, value -100 when expressions
-	Datasource    DataSourceRef `json:"datasource"`
+	DataSource    *models.DataSource `json:"datasource"`
 	JSON          json.RawMessage
 	Interval      time.Duration
 	QueryType     string
 	MaxDataPoints int64
 }
 
-type DataSourceRef struct {
-	Type string `json:"type"` // value should be __expr__
-	UID  string `json:"uid"`  // value should be __expr__
-}
-
 func (q *Query) GetDatasourceUID() string {
```

Comment:
This function is a getter for a public value with no added logic, and it's not used anywhere. I would suggest removing it.